iib/workers/tasks/build_regenerate_bundle.py

```diff
@@ -363,14 +376,65 @@ def _resolve_image_pull_specs(bundle_metadata, labels, pinned_by_iib, registry_r
             log.debug('%s will be replaced with %s', pullspec, new_pullspec.to_str())
             replacement_pullspecs[pullspec] = new_pullspec
 
+    _replace_csv_pullspecs(bundle_metadata, replacement_pullspecs, set_related_images=True)
+
+
+def _replace_image_name_from_labels(bundle_metadata, replacement_template):
+    """
+    Replace repo/image-name in the CSV pull specs with values from their labels.
+
+    :param dict bundle_metadata: the dictionary of CSV's and relatedImages pull specifications
+    :param str replacement_template: the template specifying which label values to use for
+        replacement
+    """
+    replacement_pullspecs = {}
+    label_regex = r'\{(.*?)\}'
+    replacement_labels = re.findall(label_regex, replacement_template)
+
+    for pullspec in bundle_metadata['found_pullspecs']:
+        new_pullspec = ImageName.parse(pullspec.to_str())
+        pullspec_labels = get_image_labels(pullspec.to_str())
+        modified_namespace_repo = replacement_template
+
+        for label in replacement_labels:
+            label_value = pullspec_labels.get(label)
+            if label_value:
```

Comment:
No specific reason. I was already using regex to find the label names in the string so used it to substitute too. If you feel strongly against it, I can change it.

Reply:
Why do we need regex in the first place? We should be able to create a dict that maps labels to their values, then simply call `my_string.format(**labels)`

Reply:
Ah, yes. I did think about using `.format`, but it would throw a keyError in-case there are invalid labels present and that too only for the first invalid key. In which case, we will still have to catch the exception and still use regex to find the invalid labels. So I resorted to using regex throughout. Sorry if I missed something? 

```
>>> x = '{name}-version-{version}-{something}'
>>> y = {'name': 'some_name', 'other_key':'something'}
>>> x.format(**y)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'version'
```

Reply:
You can catch `KeyError` and provide a more helpful message which could include which labels are available and the template that IIB is trying to use. That should help debugging.

I just don't think it's worth adding the regex complexity for this.